llama_index/llms/localai.py

```diff
@@ -1,25 +1,38 @@
-from typing import Any, Dict, Optional
+"""
+LocalAI is a free, open source, and self-hosted OpenAI alternative.
+
+Docs: https://localai.io/
+Source: https://github.com/go-skynet/LocalAI
+"""
+
+import warnings
+from types import MappingProxyType
+from typing import Any, Dict, Mapping, Optional
 
 from llama_index.bridge.pydantic import Field
 from llama_index.constants import DEFAULT_CONTEXT_WINDOW
 from llama_index.llms.base import LLMMetadata
 from llama_index.llms.openai import OpenAI
+from llama_index.llms.openai_like import OpenAILike
 from llama_index.llms.openai_utils import is_function_calling_model
 
-DEFAULT_API_KEY = "fake"
-DEFAULT_API_HOST = "localhost"
-DEFAULT_API_PORT = 8080
-DEFAULT_API_BASE = f"{DEFAULT_API_HOST}{DEFAULT_API_PORT}"
+# Use these as kwargs for OpenAILike to connect to LocalAIs
+DEFAULT_LOCALAI_PORT = 8080
+LOCALAI_DEFAULTS: Mapping[str, Any] = MappingProxyType(
+    {
+        "api_key": "localai_fake",
+        "api_type": "localai_fake",
+        "api_base": f"http://localhost:{DEFAULT_LOCALAI_PORT}/v1",
+    }
+)
 
 
-class LocalAI(OpenAI):
-    """
-    LocalAI is a free, open source, and self-hosted OpenAI alternative.
+def make_localai(**kwargs: Any) -> OpenAILike:
```

Comment:
Instead of a factory function, it would probably be better to just use `OpenAILike` directly no? (And also more educational, OpenAILike doesn't have any demos lol)

Reply:
Especially since this isn't a pattern used anywhere else in LlamaIndex

Reply:
Yea I understand the principle 👍 -- I just think directly using OpenAILike instead of hiding it in a wrapper makes a better example 

(again, especially because OpenAILike has zero examples too)

Reply:
It feels like this wrapper is better kept in a personal code base rather than the main wrapper, is what I'm getting at 👍

Reply:
Sorry for being nit-picky, but just trying to keep our LLM implementations consistent.

Reply:
Totally agree with the vision! Would love to have examples of OpenAILike integrating with those (vLLM is another one too)

Reply:
TIL about vLLM, so many LLM providers haha.

I have pushed this change, ready for your review again